From 399c9734ebbe3bab7a8f05143bd8e98c7efcf721 Mon Sep 17 00:00:00 2001
From: Aikar <aikar@aikar.co>
Date: Sat, 7 Oct 2017 19:54:39 -0400
Subject: [PATCH] Configurable Chunk Inhabited Timer

Vanilla stores how long a chunk has been active on a server, and
dynamically scales some aspects of vanilla gameplay to this factor.

For people who want all chunks to be treated equally, you can disable the timer.

diff --git a/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java b/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
index ed860b1..09e9fe4 100644
--- a/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
+++ b/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
@@ -267,4 +267,9 @@ public class PaperWorldConfig {
     private void firePhysicsEventForRedstone() {
         firePhysicsEventForRedstone = getBoolean("fire-physics-event-for-redstone", firePhysicsEventForRedstone);
     }
+
+    public boolean useInhabitedTime = true;
+    private void useInhabitedTime() {
+        useInhabitedTime = getBoolean("use-chunk-inhabited-timer", true);
+    }
 }
diff --git a/src/main/java/net/minecraft/world/chunk/Chunk.java b/src/main/java/net/minecraft/world/chunk/Chunk.java
index 4012467..a0e0699 100644
--- a/src/main/java/net/minecraft/world/chunk/Chunk.java
+++ b/src/main/java/net/minecraft/world/chunk/Chunk.java
@@ -1391,7 +1391,7 @@ public class Chunk {
     }
 
     public long getInhabitedTime() {
-        return this.inhabitedTime;
+        return world.paperConfig.useInhabitedTime ? this.inhabitedTime : 0; // Paper
     }
 
     public void setInhabitedTime(long newInhabitedTime) {
-- 
2.18.0

